packages/hyperion-autologging/src/ALSurfaceMutationPublisher.ts

```diff
@@ -62,10 +64,84 @@ export type InitOptions = Types.Options<
   {
     channel: ALSurfaceMutationChannel;
     cacheElementReactInfo: boolean;
+    enableDefaults?: boolean
   }
 >;
 
+function publishSurfaceInputDefaults(options: InitOptions): void {
+  const { channel } = options;
+
+  channel.addListener('al_surface_mutation_event', eventData => {
+    const { event, surface, element } = eventData;
+    if (event === 'mount_component' && surface != null && element != null) {
+      // The following is expensive, so try to jam everything we are interested in to the selector
+      const values = element.querySelectorAll<HTMLInputElement | HTMLSelectElement>('input[type=radio], input[type=checkbox], select');
+
+      if (!values.length) {
+        // Nothing to add
+        return;
+      }
+      // const metadata: Metadata = {
+      //   eventName: 'surface_input_default',
+      //   mount_event_index: "" + eventData.eventIndex,
+      //   surface,
+      // };
+      const defaults: {
+        [key: string]: {
+          type: string | null | 'radio' | 'checkbox',
+          checked: boolean,
+        } | {
+          type: 'select',
+          value: string | null,
+          text: string | null,
+        }
+      } = {};
+      for (let i = 0; i < values.length; ++i) {
+        const value = values[i];
+        const parentSurfaceNode = getAncestralSurfaceNode(value);
+        if (parentSurfaceNode === element) {
```

Comment:
So this is checking that it's the childmost surface?

That was another question,  what happens with root surface in this case,  like `/am`?  Do we get all the input/radios etc.. on the page?

Reply:
That the goal of this code, each input is logged only under its immediate ancestor surface. So if we have something like:
`<S1>
  <I1/>
  <S2>
    <I2/>
  </S2>
</S1>`

Although both I1 and I2 are selected under the S1, only I1 will be logged. I tried a bit to play with the CSS selector to try to do this there, but could not, but anyways we can improve the selector later, but this logic ensures correctness.